src/us/kbase/workspace/database/provenance/EventDate.java

```diff
@@ -0,0 +1,96 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import us.kbase.workspace.database.Util;
+
+/**
+ * The date of a specified event.
+ */
+public class EventDate {
+
+	private final String date;
+	private final String dateFormat;
+	private final String event;
+
+	public static final Pattern VALID_DATE_FORMAT_REGEX = Pattern.compile("^YYYY(-MM(-DD)?)?$");
+
+	public static final Pattern VALID_DATE_REGEX = Pattern.compile("^[12]\\d{3}(-(0[1-9]|1[0-2])(-(0[1-9]|[12]\\d|3[01]))?)?$");
+
+	private EventDate(
+		final String date,
+		final String dateFormat,
+		final String event) {
+		this.date = date;
+		this.dateFormat = dateFormat;
+		this.event = event;
+	}
+
+	/** Get the date, for example "2022-05-10".
+	 * @return the date.
+	 */
+	public String getDate() {
```

Comment:
I'm forgetting why a `Date` or better, `Instant` instance wouldn't work here

Reply:
In some cases, we only have the year; in some just the date; and in some, the full date + time.

Reply:
But it could still be represented as a Date/Instant